compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -777,3 +777,32 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         })
     }
 }
+
+impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx, PointerTag = AllocId>> InterpCx<'mir, 'tcx, M> {
+    pub fn get_alloc_from_imm_scalar_pair(
```

Comment:
Wow that is an odd function. Please add a doc comment explaining it.

The name also doesn't really reflect what it does -- it seems to specifically assume that the pair has a pointer as its first component and a length as its second component? It would probably be less complicated to start by calling `ref_to_mplace`, and then converting the *place* to an allocation-range pair. That also makes a lot more sense in general: every `MPlace` does correspond to a range in some allocation.

This function has the extra twist that the pointer *must* point to already interned memory. It might not even need `&self` and can work with just a `tcx`?

Reply:
The reason why I created this function was to get access to the `AllocRange` for the Allocation. Afaict there isn't a function that gets the `AllocRange` from an `MPlace`, so I had to get the length of the slice through the `ScalarPair`. We actually start out with an `MPlace`, so if I knew how to 'convert the place to an allocation-range pair' that would make this function obsolete and would greatly simplify the logic.